### DISCUSSION TITLE
How to access to a binding to get the convert in code behind?

### DISCUSSION THREAD

**Discussion by dbriard** [AvaloniaUI/Avalonia#13918] [Q&A] [answered]
Hi, 
I have a custom Slider from 0 to 1 that is binded to a value from 1 to 500 using a converter, how can I access to the binding from code so I can get the converter? I want to display a AutoToolTip with the value in 1-500 range in my case.
Thank for your help.

**Comment by timunie** [comment]
What is `AutoToolTip`? Can you show some sample code?

**Reply by dbriard** [reply]
Hi @timunie
It is just a feature from WPF that I reproduce in my custom slider. It display a tooltip above the thumb of the slider when dragging. This part is working.

My problem is that I added a ValueConverter to my slider (0-1) to use another scale (log scale) than the binded property (1-500).
In my custom slider, the tooltip is generated from the Value which is 0-1, and I want instead 1-500.
Currently I added another property AutoToolTipConverter where I set the same Converter that is used in the Value binding.
Instead, I would like to remove that property, and check if Value binding have a Converter, and if yes, use that converter.

I think it is possible with WPF with something like GetBindingExpression or something like that... but is there a way to get the binding on Value with Avalonia?

**Reply by timunie** [reply, marked as answer]
No that's currently not supported but we are on Binding updates. Probably will be there later. 

For now I could think of passing a Binding to the AutoTooltip. That would also provide StringFormat 😎. You can check DisplayMemberBinding on ItemsControl for example to get the syntax

**Comment by maxkatz6** [comment]
I don't think you can access the binding, so your would need to solve it in another way

**Reply by dbriard** [reply]
Thanks to both of you @maxkatz6 and @timunie, I'll continue to use a ValueConverter or some kind of formatter.
